src/pages/patientView/trialMatch/TrialMatchTable.tsx

```diff
@@ -0,0 +1,321 @@
+        return (
+            <div className={styles.tooltip}>
+                <ul className={styles.alterationUl}>
+                    {data.map((cancerType: string) => (
+                        <li>{cancerType}</li>
+                    ))}
+                </ul>
+            </div>
+        );
+    }
+
+    public mainContent(content: string) {
+        return (
+            <span>
+                <a>{content}</a>
+            </span>
+        );
+    }
+
+    public getHugoSymbolName(matches: Array<IGenomicGroupMatch>, threshold: number) {
+        const hugoSymbolSet = new Set([...matches].map(x => x.genomicAlteration.split(' ')[0].slice(1)));
+        if (hugoSymbolSet.size <= threshold) {
+            return [...hugoSymbolSet].join(', ');
+        }
+        return `${hugoSymbolSet.size} genes`;
+    }
+
+    public buildDetailedTrialMatches() {
+        const groupByTrialId = _.groupBy(this.props.trialMatches, 'nctId');
+        let matchedTrials: Array<IDetailedTrialMatch> = [];
+        _.forEach(groupByTrialId, (trialGroup, trialId) => {
+            let matchedTrial:any = _.find( this.props.trials, { 'nctId': trialId } );
+            matchedTrial['priority'] = 0; // highest priority
+            matchedTrial['matches'] = [];
+            const groupByArm = _.groupBy(trialGroup, 'armDescription');
+            _.forEach(groupByArm, (armGroup, armDescription) => {
+                let armMatch: IArmMatch = {
+                    armDescription: armDescription,
+                    drugs: [],
+                    matches: []
+                };
+                const isControlArm = getDrugsAndExcludeControlArms(armDescription, matchedTrial['treatmentList']['step'][0]['arm'], armMatch);
+                if (isControlArm) return; // skip control arm
+
+                const groupByAge = _.groupBy(armGroup, 'trialAgeNumerical');
+                _.forEach(groupByAge, (ageGroup, age) => {
+                    const cancerTypes =  _.uniq(_.map(ageGroup, 'trialOncotreePrimaryDiagnosis'));
+                    let generalCancerTypes: Array<string> = [];
+                    let notCancerTypes: Array<string> = [];
+                    _.map(cancerTypes, (item) => {
+                       if (item.indexOf('!') > -1) {
+                           notCancerTypes.push(item.replace('!', ''));
+                       } else {
+                           generalCancerTypes.push(item);
+                       }
+                    });
+                    let clinicalGroupMatch: IClinicalGroupMatch = {
+                        trialAgeNumerical: age,
+                        trialOncotreePrimaryDiagnosis: {
+                            general: generalCancerTypes,
+                            not: notCancerTypes
+                        },
+                        matches: [],
+                        notMatches: []
+                    };
+                    const groupByGenomicAlteration = _.groupBy(groupByAge[age], 'genomicAlteration');
```

Comment:
shortcut

Reply:
Changed to `const matchesGroupedByGenomicAlteration = _.groupBy(matchesGroupedByAge[age], (trial: ITrialMatch) => trial.genomicAlteration);`.